### ISSUE TITLE
lib/prawn/text.rb - consider deleting '#' characters 

### ISSUE THREAD

**Issue by ghost** [prawnpdf/prawn#807]
Hi,

Not sure if sandal considers this good or not - if not it can just be closed. Thanks.

Currently you can do this code:

```
text 'Ruby in Red', :color => 'FF0000'
text 'Ruby in Red', :color => '#FF0000'
```

The first variant works fine, and displays the text in red.

The second line displays the code in green (???)

Screenshot:
![alt text](http://i.imgur.com/beq9Z7D.png)

My proposals:
(a) Leading '#' characters will be ignored (this behaviour should be documented). The effect would be that the second line would be treated like the first line (unless perhaps there was a reason for this which I overlooked, but to me right now this seems more like a tiny bug)
(b) The green colour here, I am not sure if it makes sense. I assume that currently the '#' marks this as invalid? In that event, perhaps it should default to black? (e. g. black text on white background, by default)

**Comment by practicingruby** [comment]
@packetmonkey: Do you still want to work on this? Since it's a small patch, I could probably write it myself if you don't plan to do it in the next few days, so that we can include it in the 2.0.0 release.

**Comment by packetmonkey** [comment]
This totally dropped off my radar, I'll try to knock it out this weekend but I won't feel hurt if you beat me to it.

**Comment by practicingruby** [comment]
I'll drop a note on this ticket if I start working on this, otherwise go ahead and give it a shot.

**Comment by packetmonkey** [comment]
Going to assign the issue to myself so it shows up on my GH dashboard.

**Comment by practicingruby** [comment]
I realized that this isn't a release blocker because we'd just be turning an undefined behavior into an exception. So rather than rushing to build a fix before the 2.0.0 release (which may happen today or tomorrow), just fix this when you get around to it @packetmonkey.

**Comment by shevegen** [comment]
Gregory, it is time to say it - I <3 your work on prawn. :)  (There was an older "port", ruby-freepdf (fpdf) loosely based on the php-pdf, but it never was high quality code, very difficult to navigate and modify; prawn is much much better from a technical point of view)

**Comment by practicingruby** [comment]
Thanks @shevegen. Prawn is very much the work of dozens of contributors. Most of the good ideas aren't mine. :-P

**Comment by petergoldstein** [comment]
@pointlessone This issue should be closed, as a behavioral decision was made and merged as discussed in #869
